File: AspNetMvcUnleashed/0672329980_source/CommonCode/MvcFakes/MvcFakes/FakeIdentity.cs
using System;
using System.Security.Principal;

namespace MvcApplication1.Tests.Models
{
    public class FakeIdentity : IIdentity
    {
        private string _name;

        public FakeIdentity(string name)
        {
            _name = name;
        }

        #region IIdentity Members

        public string AuthenticationType
        {
            get { throw new NotImplementedException(); }
        }

        public bool IsAuthenticated
        {
            get { return !string.IsNullOrEmpty(_name); }
        }

        public string Name
        {
            get { return _name; }
        }

        #endregion
    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter12/VB/MvcApplication1/MvcApplication1.Tests/Models/FakeIdentity.vb
Imports System.Security.Principal

Public Class FakeIdentity
    Implements IIdentity

    Private _name As String

    Public Sub New(ByVal name As String)
        _name = name
    End Sub

#Region "IIdentity Members"

    Public ReadOnly Property AuthenticationType() As String Implements IIdentity.AuthenticationType
        Get
            Throw New NotImplementedException()
        End Get
    End Property

    Public ReadOnly Property IsAuthenticated() As Boolean Implements IIdentity.IsAuthenticated
        Get
            Throw New NotImplementedException()
        End Get
    End Property

    Public ReadOnly Property Name() As String Implements IIdentity.Name
        Get
            Return _name
        End Get
    End Property

#End Region

End Class
